### DISCUSSION TITLE
Using useSWRInfinite the data of the first index comes again and again

### DISCUSSION THREAD

**Discussion by Leoahsan** [vercel/swr#1639] [Q&A]
After making this hook using infinity scroll, when I am scrolling the first scroll, the data of page 1 comes and in the second scroll, the data of page 2 comes, but in the 3rd scroll, the data of page 1 comes again and this problem comes continuously like - after page 3. Again, after page 1 data comes and again after page 4, the data of page 1 comes again like this, 
So please help me out what is the problem here on these codes... please

The hook code is like this ...

	`import useSWRInfinite from "swr/infinite";
        export const useInfiniteScroll = (url, option) => {
        const POST_LIMIT = 2;
	    const getKey = (pageIndex, previousPageData) => {
		    pageIndex = pageIndex + 1;
		    if (previousPageData && !previousPageData.length) return null; // reached the end
		    return `${url}?_start=${pageIndex}&_limit=${POST_LIMIT}`; // SWR key
	    };

	const {
		data,
		size: page,
		setSize: setPage,
		error,
		mutate,
		isValidating,
	} = useSWRInfinite(getKey, option);

	const scrolledData = [].concat.apply([], data);

	const isLoadingMore = data && typeof data[page - 1] === "undefined";

	const isReachedEnd = data && data[data.length - 1]?.length < POST_LIMIT;
	return {
		data,
		mutate,
		error,
		page,
		setPage,
		scrolledData,
		isLoadingMore,
		isReachedEnd,
		isValidating,
	}; };`

And the component code is like this...

     `const dataFromApi = useApiData("/home/index/");
    const {
        scrolledData: paginatedPosts,
        page,
        setPage,
        isReachedEnd,
    } = useInfiniteScroll(dataFromApi);

    return (
        <HomeLayout>
            {paginatedPosts.length !== 0 && paginatedPosts !== undefined && (
                <InfiniteScroll
                    dataLength={paginatedPosts?.length}
                    next={() => setPage(page + 1)}
                    hasMore={!isReachedEnd}
                    loader={<HomePlaceholder />}
                    endMessage={<NoDataView />}>
                    {paginatedPosts.map((singlePost, index) => (
                        <div className="card is-post" key={index}>
                            <span key={index}>{singlePost.title}</span>
                        </div>
                    ))}
                </InfiniteScroll>
            )}
        </HomeLayout>
    );`

**Comment by Hmoulvad** [comment]
set revalidateFirstPage to false.
